internal/mux/mux_test.go

```diff
@@ -140,3 +140,21 @@ func TestNoReader(t *testing.T) {
 		t.Errorf("got: %v, want: %v", buf, make([]byte, len(buf)))
 	}
 }
+
+func TestEmptyBuffersDontBlock(t *testing.T) {
+	w1 := bytes.Buffer{}
+	w2 := bytes.Buffer{}
+	m := mux.New(1, 1)
+	m.AddSource(&w1)
+	m.AddSource(&w2)
+
+	w1.Write(make([]byte, 100))
+	buf := make([]byte, 100)
+	n, err := m.Read(buf)
```

Comment:
Use `io.ReadFull`

Reply:
Alternative title for this PR: Diane learns about `io.ReadFull` and then learns it solves every problem ever.

Will change :)